backend/siarnaq/bracket/challonge.py

```diff
@@ -126,21 +126,118 @@ def get_tournament_data(tournament: Tournament, *, is_private: bool):
 
     r = requests.get(url, headers=_headers)
     r.raise_for_status()
-    return r.json()
+    data = r.json()
 
+    # Handle bracket-reset finals by internally splitting it up into two rounds,
+    # which Challonge does not do by default.
+    from siarnaq.api.episodes.models import TournamentStyle
+
+    if tournament.style == TournamentStyle.DOUBLE_ELIMINATION:
+        # Find the very last match (by play order),
+        # which is the second match of finals
+        # in case of bracket reset
+        matches = [item for item in data["included"] if item["type"] == "match"]
+        match_last = max(
+            matches, key=lambda match: match["attributes"]["suggestedPlayOrder"]
+        )
+        # Give it its own round
+        match_last["attributes"]["round"] += 1
+
+    return data
 
-def get_round_indexes(tournament: Tournament, *, is_private: bool):
+
+# TODO should this be private-only, and thus should I drop the is_private flag?
+# Could use help/opinions
+def _get_round_indexes(tournament: Tournament, *, is_private: bool):
+    """
+    Returns round indexes of the tournament,
+    in order of Challonge's suggested play order.
+    """
     tournament_data = get_tournament_data(tournament, is_private=is_private)
 
-    round_indexes = set()
-    for item in tournament_data["included"]:
-        match item:
-            case {"type": "match", "attributes": {"round": round_index}}:
-                round_indexes.add(round_index)
+    round_indexes = list()
+
+    matches = [item for item in tournament_data["included"] if item["type"] == "match"]
+    matches.sort(key=lambda i: i["attributes"]["suggestedPlayOrder"])
+
+    for match in matches:
+        round_index = match["attributes"]["round"]
+        if round_index not in round_indexes:
+            round_indexes.append(round_index)
 
     return round_indexes
 
 
+def _get_round_indexes_and_names(tournament: Tournament, *, is_private=bool):
```

Comment:
always default to private? is that odd

Reply:
This function should probably keep `is_private` around, since it queries a specific bracket. Also fix typo:
```suggestion
def _get_round_indexes_and_names(tournament: Tournament, *, is_private: bool):
```